### PR TITLE
[W5][W09-2]Jin Jing

### PR DESCRIPTION
**Pull request by jing1324**
Please review thanks.

### CODE REVIEW

**Comment by jing1324** [comment]
I am not sure how to do the Junit test for this command. It is similar to the help and list command for which both do not have Junit tests.

**Review by Lookuz — COMMENTED** [review]
I think that the history command is really useful in tracking previous commands in order to backtrack what progress was made. One suggestion I would make would be to further add on to the history command by making use of history command to implement undo/redo previous commands.

**Comment by daviddl9** [comment]
Certainly a useful command to have. Perhaps you can consider keeping a limit of the maximum number of past commands to keep track of. Storing millions of commands (perhaps in one long run of the addressbook) can cause slower performance due to high memory usage.

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -28,6 +30,7 @@
     private TextUi ui;
     private StorageFile storage;
     private AddressBook addressBook;
+    public static ArrayList<String> History = new ArrayList<>();
```
Comment (marlenekoh) [inline src/seedu/addressbook/Main.java:33]: Can consider using a stack to store history as well (:

File: src/seedu/addressbook/commands/HistoryCommand.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.Main;
+
+
+
```
Comment (marlenekoh) [inline src/seedu/addressbook/commands/HistoryCommand.java:6]: Remove unecessary line breaks

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (marlenekoh) [inline src/seedu/addressbook/parser/Parser.java:14]: Wildcard imports violate the coding standard. Please use single imports and change your settings in Intellij: `Settings` > `Editor` > `Code Style` > `Java` > `Imports` and set both class count and names count to 99 (or any large number)

**Review by marlenekoh — APPROVED** [review]
@jing1324 Good job at updating your documentation, help command and tests (: Regarding the Junit Test for this command,  you can execute some commands and ensure that these commands are being added into your ArrayList correctly (i.e. assert that the String is correct).

Overall, good effort (: Please close this PR after reading review comments.

**Pull request closed without merging**